python/nav/ipdevpoll/plugins/juniperalarm.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (C) 2022 Sikt
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""Juniper chassis/system alarms"""
+
+from twisted.internet import defer
+from twisted.internet.defer import returnValue
+
+from django.db import transaction
+
+from nav.event2 import EventFactory
+from nav.mibs.juniper_alarm_mib import JuniperAlarmMib
+from nav.models import event
+from nav.ipdevpoll import Plugin, db
+
+YELLOW_EVENT = EventFactory(
+    'ipdevpoll', 'eventEngine', 'juniperYellowAlarmState', 'juniperYellowAlarmNonZero', 'juniperYellowAlarmZero'
+)
+RED_EVENT = EventFactory(
+    'ipdevpoll', 'eventEngine', 'juniperRedAlarmState', 'juniperRedAlarmNonZero', 'juniperRedAlarmZero'
+)
+
+
+class JuniperChassisAlarm(Plugin):
+    """Retrives the number of red and yellow alarms in a chassis
+
+    This is done by attempting to retrieve the gauges in JUNIPER-ALARM-MIB:
+
+    * jnxYellowAlarmCount
+    * jnxRedAlarmCount
+
+    If the count is not zero, a start event for that color is sent. When the
+    count goes back to zero, an end-event is sent.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.mib = JuniperAlarmMib
+        super().__init__(*args, **kwargs)
+
+    @defer.inlineCallbacks
+    def handle(self):
+        self._logger.debug("Collecting any juniper led alarms")
+
+        self.yellow_count = yield self.mib.get_yellow_alarm_count()
+        self.red_count = yield self.mib.get_red_alarm_count()
+
+        if self.yellow_count:
+            yield self._yellow_alarm_count_non_zero(self)
+        else:
+            yield self._yellow_alarm_count_zero(self)
+
+        if self.red_count:
+            yield self._red_alarm_count_non_zero(self)
+        else:
+            yield self._red_alarm_count_zero(self)
+
+    @transaction.atomic()
+    def _yellow_alarm_count_non_zero(self):
+        YELLOW_EVENT.start(None, self.netbox.id).save()
```

Comment:
Rather than the opaque use of `None` as the first argument here (and the other event posting code lines), it would be more readable as:
```suggestion
        YELLOW_EVENT.start(netbox=self.netbox.id).save()
```